plugin/src/main/java/org/opensearch/ml/action/models/GetModelTransportAction.java

```diff
@@ -63,37 +70,54 @@ public GetModelTransportAction(
         TransportService transportService,
         ActionFilters actionFilters,
         Client client,
+        SdkClient sdkClient,
         Settings settings,
         NamedXContentRegistry xContentRegistry,
         ClusterService clusterService,
-        ModelAccessControlHelper modelAccessControlHelper
+        ModelAccessControlHelper modelAccessControlHelper,
+        MLFeatureEnabledSetting mlFeatureEnabledSetting
     ) {
         super(MLModelGetAction.NAME, transportService, actionFilters, MLModelGetRequest::new);
         this.client = client;
+        this.sdkClient = sdkClient;
         this.settings = settings;
         this.xContentRegistry = xContentRegistry;
         this.clusterService = clusterService;
         this.modelAccessControlHelper = modelAccessControlHelper;
+        this.mlFeatureEnabledSetting = mlFeatureEnabledSetting;
     }
 
     @Override
     protected void doExecute(Task task, ActionRequest request, ActionListener<MLModelGetResponse> actionListener) {
         MLModelGetRequest mlModelGetRequest = MLModelGetRequest.fromActionRequest(request);
         String modelId = mlModelGetRequest.getModelId();
+        String tenantId = mlModelGetRequest.getTenantId();
+        if (!TenantAwareHelper.validateTenantId(mlFeatureEnabledSetting, tenantId, actionListener)) {
+            return;
+        }
         FetchSourceContext fetchSourceContext = getFetchSourceContext(mlModelGetRequest.isReturnContent());
+        GetDataObjectRequest getDataObjectRequest = new GetDataObjectRequest.Builder()
+            .index(ML_MODEL_INDEX)
+            .id(modelId)
+            .fetchSourceContext(fetchSourceContext)
+            .build();
         GetRequest getRequest = new GetRequest(ML_MODEL_INDEX).id(modelId).fetchSourceContext(fetchSourceContext);
         User user = RestActionUtils.getUserContext(client);
         boolean isSuperAdmin = isSuperAdminUserWrapper(clusterService, client);
 
         try (ThreadContext.StoredContext context = client.threadPool().getThreadContext().stashContext()) {
-            ActionListener<MLModelGetResponse> wrappedListener = ActionListener.runBefore(actionListener, () -> context.restore());
+            ActionListener<MLModelGetResponse> wrappedListener = ActionListener.runBefore(actionListener, context::restore);
             client.get(getRequest, ActionListener.wrap(r -> {
```

Comment:
Use sdkClient to get model information

Reply:
As we discussed yesterday offline, after completing multi-tenancy part, then I'll come back to use `sdkClient` for get, create, delete (in addition with proper exception handling).